### PR TITLE
Added quick rebalance integration test

### PR DESCRIPTION
**Pull request by mhowlett**
this test hangs in v1.8.2, as indicated.

### CODE REVIEW

**Comment by mhowlett** [comment]
actual code used to create the logs (should be functionally same as test code):

```
        private static IConsumer<Null, string> quickRebalanceCreateConsumer(
            String bootstrapServers, String groupId)
        {
            return new ConsumerBuilder<Null, string>(
                new ConsumerConfig
                {
                    BootstrapServers = bootstrapServers,
                    GroupId = groupId,
                    AutoOffsetReset = AutoOffsetReset.Earliest,
                    Debug = "all",
                    EnableAutoCommit = true,
                    EnableAutoOffsetStore = false,
                    AutoCommitIntervalMs = 500,
                    PartitionAssignmentStrategy = PartitionAssignmentStrategy.RoundRobin,
                })
                .SetPartitionsAssignedHandler((c, partitions) => {
                    Console.Error.WriteLine(
                        "Partitions assigned: [" +
                        string.Join(',', partitions.Select(p => p.Partition.Value)) +
                        "]");
                })
                .SetPartitionsRevokedHandler((c, partitions) => {
                    var remaining = c.Assignment.Where(atp => partitions.Where(rtp => rtp.TopicPartition == atp).Count() == 0);
                    Console.Error.WriteLine(
                        "Partitions revoked: [" +
                        string.Join(',', partitions.Select(p => p.Partition.Value)) +
                        "]");
                })
                .Build();
        }

        private static void LogErr(String msg)
        {
            Console.Error.WriteLine(msg);
            Console.Error.Flush();
        }

        public static void Test3()
        {
            string bootstrapServers = "localhost:9092";
            var groupId = Guid.NewGuid().ToString();
            var numPartitions = 4;
            using (var topic = new TemporaryTopic(bootstrapServers, numPartitions))
            {
                using (var producer = new ProducerBuilder<Null, string>(new ProducerConfig { BootstrapServers = bootstrapServers }).Build())
                {
                    for (int i=0; i<numPartitions; ++i)
                    {
                        for (int j=0; j<100; ++j)
                        {
                            producer.Produce(new TopicPartition(topic.Name, i), new Message<Null, string> { Value = "test" });
                        }
                    }
                    producer.Flush();
                }

                LogErr("---- Create /1");
                var consumer1 = quickRebalanceCreateConsumer(bootstrapServers, groupId);
                LogErr("---- Subscribe /1");
                consumer1.Subscribe(topic.Name);
                ConsumeResult<Null, String> cr = null;
                while (cr == null || cr.Message == null)
                {
                    LogErr("---- Consume /1");
                    cr = consumer1.Consume(TimeSpan.FromSeconds(2));
                }

                LogErr("---- Create /2");
                var consumer2 = quickRebalanceCreateConsumer(bootstrapServers, groupId);
                LogErr("---- Subscribe /2");
                consumer2.Subscribe(topic.Name);
                cr = null;
                while (cr == null || cr.Message == null)
                {
                    LogErr("---- Consume /2");
                    cr = consumer2.Consume(TimeSpan.FromSeconds(2));
                }

                LogErr("---- Close /1");
                consumer1.Close();
                cr = null;
                while (cr == null || cr.Message == null)
                {
                    LogErr("---- Consume /2");
                    cr = consumer2.Consume(TimeSpan.FromSeconds(2));
                }

                LogErr("---- Close /2");
                Console.Error.Flush();
                consumer2.Close();
            }
        }
```

File: test/Confluent.Kafka.IntegrationTests/Tests/Consumer_QuickRebalance.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                {
+                    BootstrapServers = bootstrapServers,
+                    GroupId = groupId,
+                    AutoOffsetReset = AutoOffsetReset.Earliest,
+                    EnableAutoCommit = true,
+                    EnableAutoOffsetStore = false,
+                    AutoCommitIntervalMs = 500,
+                    PartitionAssignmentStrategy = PartitionAssignmentStrategy.RoundRobin,
+                })
+                .SetPartitionsAssignedHandler((c, partitions) => { })
```
Comment (mhowlett) [inline test/Confluent.Kafka.IntegrationTests/Tests/Consumer_QuickRebalance.cs:41]: note, in C#, assign is not expected (or allowed) here.

Comment (3schwartz) [reply test/Confluent.Kafka.IntegrationTests/Tests/Consumer_QuickRebalance.cs:41]: Hi @mhowlett 
What do you mean by not using assign from this handler - isn't it supported to return the exact topic partitions from this handler and then these will be used by the consumer?

File: test/Confluent.Kafka.IntegrationTests/Tests/Consumer_QuickRebalance.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                while (cr == null || cr.Message == null)
+                {
+                    cr = consumer1.Consume(TimeSpan.FromSeconds(2));
+                }
+                var consumer2 = quickRebalanceCreateConsumer(bootstrapServers, groupId);
+                consumer2.Subscribe(topic.Name);
+                cr = null;
+                while (cr == null || cr.Message == null)
+                {
+                    // Should not get stuck in this loop.
```
Comment (edenhill) [inline test/Confluent.Kafka.IntegrationTests/Tests/Consumer_QuickRebalance.cs:73]: Is there a test framework timeout that makes sure it doesn't get stuck here?

**Review by edenhill — APPROVED** [review]

**Pull request closed without merging**